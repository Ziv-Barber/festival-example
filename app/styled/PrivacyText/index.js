// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import styled from 'styled-components'

// Material-ui stuff:
import MuiTypography from '@material-ui/core/Typography'

// Our types:
import type { PrivacyTextPropsType } from './index.types'

/**
 * PrivacyText react component.
 * @reactProps {string} text Parameters for the component.
 * @return New element.
 */
const PrivacyText = (props: PrivacyTextPropsType) => (
  <MuiTypography
    className={props.className || 'privacyText'}
    variant="h5"
    component="h2"
  >
    {props.children}
  </MuiTypography>
)

export default styled(PrivacyText)`
  margin: 0 !important;
  padding: ${(props) => props.theme.spacing.unit * 2}px !important;
  background: ${(props) => props.theme.palette.secondary.dark};
`

// In case that someone need our types:
export type { PrivacyTextPropsType as Props }
