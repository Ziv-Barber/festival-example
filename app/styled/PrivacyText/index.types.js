// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars

export type PrivacyTextPropsType = {
  /** Class for the root element. */
  className?: string,
  children?: React.Node
}
