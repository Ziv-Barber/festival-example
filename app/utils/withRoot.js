// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import CssBaseline from '@material-ui/core/CssBaseline'
import theme from '../theme'

/**
 * withRoot HOC to wrap a react component with the app's theme provider.
 */
const withRoot = (Component: React.ComponentType<any>) => {
  const withRootBase = (props: any) => (
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <Component {...props} />
        </React.Fragment>
      </ThemeProvider>
    </MuiThemeProvider>
  )

  // $FlowFixMe
  withRootBase.displayName = `withRoot(${Component.displayName ||
    Component.name})`

  return withRootBase
}

export default withRoot
