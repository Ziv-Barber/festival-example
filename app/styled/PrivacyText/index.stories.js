import React from 'react' // eslint-disable-line no-unused-vars
import { storiesOf } from '@storybook/react'
import PrivacyText from './index'

storiesOf('PrivacyText', module).add(
  'basic',
  () => <PrivacyText text="Basic example" />,
  { notes: { markdown: 'This is an example to the PrivacyText component.' } }
)
