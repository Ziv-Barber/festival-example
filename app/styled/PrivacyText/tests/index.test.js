// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import PrivacyText from '../index'
import withRoot from '../../../utils/withRoot'

// For flow:
let { describe, it, afterEach, expect } = global

/** @test {PrivacyText} */
describe('<PrivacyText />', () => {
  afterEach(cleanup)

  /** @test {PrivacyText} */
  it('should match snapshort', () => {
    const CompToTest = () => <PrivacyText>This is a text</PrivacyText>
    const App = withRoot(CompToTest)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })

  /** @test {PrivacyText} */
  it('should render', () => {
    const CompToTest = () => <PrivacyText>This is a text</PrivacyText>
    const App = withRoot(CompToTest)
    const { getByText } = render(<App />)
    expect(getByText(/^This is a text/)).not.toBeNull()
  })
})
