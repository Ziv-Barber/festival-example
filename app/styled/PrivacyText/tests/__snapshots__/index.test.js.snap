// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivacyText /> should match snapshort 1`] = `
<div>
  <h2
    class="MuiTypography-root-1 MuiTypography-h5-17 PrivacyText-rns7yr-0 hRsTIG"
  >
    This is a text
  </h2>
</div>
`;
